src/js/components/FrameworkConfigurationForm.js

```diff
@@ -0,0 +1,315 @@
+import React, { Component, PropTypes } from "react";
+import classNames from "classnames";
+import DefaultTitleField
+  from "react-jsonschema-form/lib/components/fields/TitleField";
+import SchemaForm from "react-jsonschema-form";
+import TabButton from "#SRC/js/components/TabButton";
+import TabButtonList from "#SRC/js/components/TabButtonList";
+import Tabs from "#SRC/js/components/Tabs";
+import Util from "#SRC/js/utils/Util";
+import JSONEditor from "#SRC/js/components/JSONEditor";
+import PageHeaderNavigationDropdown
+  from "#SRC/js/components/PageHeaderNavigationDropdown";
+import UniversePackage from "#SRC/js/structs/UniversePackage";
+import CosmosErrorMessage from "#SRC/js/components/CosmosErrorMessage";
+import SchemaField from "#SRC/js/components/SchemaField";
+
+export default class FrameworkConfigurationForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      errorSchema: null
+    };
+  }
+
+  getFormattedSectionLabel(label) {
```

Comment:
label = "" to check for null or undefined state?

Reply:
@bstavroulakis please note that defaults only apply if the value is undefined.

```js
const { a = "default", b = "default" } = { a: undefined, b: null};

console.log(a) //  default
console.log(b) //  null
```

Reply:
The `label` will not be undefined or null. This method is directly called for property names from the JSON schema. I would prefer to avoid this check, I believe it's overly defensive.